### PR TITLE
ignore failing `inotify_add_watch` unless explicitly asked to watch

### LINKED ISSUE #85: Linux: `inotify_add_watch` failed when creating and deleting folders
This is a variant of https://github.com/parcel-bundler/watcher/issues/76 with a different repro. There seems to be a possible race condition between detection a new folder that get's added to the watched root, the resulting call to `inotify_add_watch` and the folder getting deleted at the same time. If this happens, the watcher terminates with `Error: inotify_add_watch on '<path>' failed: No such file or directory`

Repro:

```js
const watcher = require('./index');
const fs = require('fs');

let firstDeleteThenCreate = false;

(async function () {
    let subscription = await watcher.subscribe('<path>', (err, events) => {
        console.log(1, err, events);
    }).catch(err => {
        console.error(2, err);
    }).then(async () => {

        for (let i = 0; i < 1000; i++) {
            await fs.promises.mkdir('<path>/folder');
            await fs.promises.rmdir('<path>/folder');
        }

        console.log('Done');
    });
})();
```

Instead of throwing an error here:

https://github.com/parcel-bundler/watcher/blob/fa1b33fbb626f1e406dd1cb174676de9025f8d5f/src/linux/InotifyBackend.cc#L79

I think it would be better to ignore this event and remove the path from the tree.

### PR DESCRIPTION
**Pull request by bpasero**
This PR fixes #85 by gracefully handling failing `inotify_add_watch` calls instead of terminating.

### CODE REVIEW

File: src/linux/InotifyBackend.cc [hunk 1]
```diff
@@ -165,7 +170,10 @@ bool InotifyBackend::handleSubscription(struct inotify_event *event, std::shared
     DirEntry *entry = sub->tree->add(path, CONVERT_TIME(st.st_mtim), S_ISDIR(st.st_mode));
 
     if (entry->isDir) {
-      watchDir(*watcher, entry, sub->tree);
+      bool success = watchDir(*watcher, entry, sub->tree);
+      if (!success) {
+        sub->tree->remove(path);
+      }
```
Comment (bpasero) [inline src/linux/InotifyBackend.cc:176]: I was not entirely sure if I should `return false` after this call. It looks like this drives the call to `watchers.insert((*it)->watcher);`.

Comment (DeMoorJasper) [reply src/linux/InotifyBackend.cc:176]: Pretty sure it should return false, but @devongovett probably knows better

Comment (bpasero) [reply src/linux/InotifyBackend.cc:177]: I added a `return false` now.

**Comment by bpasero** [comment]
@devongovett I think this is ready for a review now, I tested my changes on all platforms and cannot reproduce the CI failures on macOS. And my changes should not have an impact on macOS anyway.

**Review by devongovett — APPROVED** [review]

**Pull request merged**